ntp-proto/src/time_types.rs

```diff
@@ -235,6 +235,33 @@ impl NtpDuration {
         .to_be_bytes()
     }
 
+    #[cfg(feature = "ntpv5")]
+    pub(crate) const fn from_bits_time32(bits: [u8; 4]) -> Self {
+        NtpDuration {
+            duration: (u32::from_be_bytes(bits) as i64) << 4,
+        }
+    }
+
+    #[cfg(feature = "ntpv5")]
+    pub(crate) const fn to_bits_time32(self) -> [u8; 4] {
+        // serializing negative durations should never happen
+        // and indicates a programming error elsewhere.
+        // as for duration that are too large, saturating is
+        // the safe option.
+        assert!(self.duration >= 0);
+
+        // Other then for short implementation this might very well overflow sometimes, we just saturate in that case
+        const MAX_TIME32: NtpDuration = NtpDuration {
+            duration: 0x00_00_00_0F_FF_FF_FF_F0,
+        };
+
+        match self.duration > MAX_TIME32.duration {
```

Comment:
With the change below this could also be simplified to:
```suggestion
        match self.duration > 0x0F_FF_FF_FF_F0 {
```